modules/core/src/v2/coins/sol.ts

```diff
@@ -100,8 +99,35 @@ export class Sol extends BaseCoin {
   }
 
   // TODO (https://bitgoinc.atlassian.net/browse/STLX-10202)
-  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    throw new MethodNotImplementedError('verifyTransaction method not implemented');
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<any> {
+    let totalAmount = 0;
+    const coinConfig = coins.get(this.getChain());
+    const { txParams: txParams, txPrebuild: txPrebuild } = params;
+    const transaction = new accountLib.Sol.Transaction(coinConfig);
+    if (!txPrebuild.txBase64) {
+      throw new Error('missing required tx prebuild property txBase64');
+    }
+    transaction.fromRawTransaction(txPrebuild.txBase64);
+    const explain = transaction.explainTransaction();
+    explain.outputs.should.deepEqual(txParams.recipients);
+    const transactionJson = transaction.toJson();
+    if (txParams.memo != explain.memo) {
+      throw new Error('Diferent memo');
```

Comment:
Different memo.

Reply:
Message should be like this:
Explained tx memo does not match with expected txParams memo